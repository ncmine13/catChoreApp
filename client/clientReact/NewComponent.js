var React = require('react');

class NewComponent extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			roommatename: '',
			chorename: '',
		}
	}
	handleNameInput(e){
		this.setState({roommatename: e.target.value});
	}
	handleSelectChange(e){
		this.setState({chorename: e.target.value});
	}
	handleDataEntry(e){
		e.preventDefault();
		this.props.createItem(this.state);
	}
	render() {
		return (
			<div className="inputComponentDiv">
				<form onSubmit={this.handleDataEntry.bind(this)}>
					<div>
						<input type="text" value={this.state.roommatename} onChange={this.handleNameInput.bind(this)}/>
					</div>
					<div>
						<select value={this.state.value} onChange={this.handleSelectChange.bind(this)}>
							<option>Select</option>
							<option value="cleaned the litter box">cleaned litter box</option>
							<option value="bought food">bought food</option>
							<option value="trimmed the kitty nails">trimmed nails</option>
							<option value="fed the kitty">fed the kitty</option>
						</select>
					</div>
					<div>
						<button type="submit"> Log :) </button>
					</div>
				</form>
			</div>
		)
	}
}

module.exports = NewComponent;