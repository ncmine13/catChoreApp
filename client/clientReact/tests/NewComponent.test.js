import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json'
import NewComponent from '../NewComponent';


test('NewComponent should render as expected', () => {
	const component = shallow(<NewComponent />)
	const tree = toJson(component)
	console.log(tree)
	expect(tree).toMatchSnapshot();
})

//should render 
//should set state based on entries in input fields
//should send state up to input component when log button is pressed