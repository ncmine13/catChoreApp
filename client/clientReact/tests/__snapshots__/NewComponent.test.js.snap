// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewComponent should render as expected 1`] = `
<div
  className="inputComponentDiv"
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <select
        onChange={[Function]}
      >
        <option>
          Select
        </option>
        <option
          value="cleaned the litter box"
        >
          cleaned litter box
        </option>
        <option
          value="bought food"
        >
          bought food
        </option>
        <option
          value="trimmed the kitty nails"
        >
          trimmed nails
        </option>
        <option
          value="fed the kitty"
        >
          fed the kitty
        </option>
      </select>
    </div>
    <div>
      <button
        type="submit"
      >
         Log :) 
      </button>
    </div>
  </form>
</div>
`;
